### PR TITLE
add method for creating ZOS VM on OpenvCloud env

### PR DESCRIPTION
**Pull request by katia-e**
#### What this PR resolves:
Create method to create ZOS VM connected to given ZeroTier network and returne both machine object and ZeroTier network information.

### CODE REVIEW

File: JumpScale9Lib/clients/openvcloud/Space.py [hunk 1]
```diff
@@ -339,6 +339,112 @@ def machine_create(
...
+            - zerotier_client (required): name of ZeroTier client instance.
+            - memsize (defaults to 2): memory size in MB or in GB, e.g. 4096.
+            - vcpus (defaults to 1): number of virtual CPU cores; value is ignored in versions prior to 3.x, use sizeId in order to set the number of virtual CPU cores.
+            - disksize (default to 10): boot disk size in MB.
+            - image (defaults to "IPXE Boot"): name of the Zero OS image to load.
+            - sizeId (optional): overrides the value set for memsize, denotes the type or "size" of the virtual machine, actually sets the number of virtual CPU cores and amount of memory, see the sizes property of the cloud space for the sizes available in the cloud space.
+            - stackId (optional): identifies the grid node on which to create the virtual machine, if nothing specified (recommended) OpenvCloud will decide where to create the virtual machine.
+            - description (optional): machine description.
+            - timeout (defaults to 20): timeout on waiting node to join ZeroTier network.
+            - branch (defaults to "master"): zero-os branch.
```
Comment (ahussein) [inline JumpScale9Lib/clients/openvcloud/Space.py]: as far as i can see the branch will default to development not master

File: JumpScale9Lib/clients/openvcloud/Space.py [hunk 2]
```diff
@@ -339,6 +339,112 @@ def machine_create(
...
+        Raises:
+            - RuntimeError if machine with given name already exists.
+            - RuntimeError if machine name contains spaces
+            - RuntimeError if machine name contains underscores
+        """
+
+        # url pointing to a (ipxe)[http://ipxe.org/scripting/] script.
+        ipxe = 'https://bootstrap.gig.tech/ipxe/development/{zerotier_id}/organisation={organization}%20{branch}'.format(
```
Comment (ahussein) [inline JumpScale9Lib/clients/openvcloud/Space.py]: any reason why the branch variable is not used here ?

File: JumpScale9Lib/clients/openvcloud/Space.py [hunk 3]
```diff
@@ -339,6 +339,112 @@ def machine_create(
...
+                    response['config']['nodeId'], zerotier_id))
+
+        # check that accepted VM belongs to the vdc
+        if response['physicalAddress'] != machine.ipaddr_public:
+            # disconnect VM
+            candidate['config']['authorized'] = False
+            zerotier.client.network.updateMember(candidate, candidate['nodeId'], zerotier_id)
+            raise RuntimeError('Node with unexpected physical address %s tried to connect to ZeroTier network %s' %(
+                response['physicalAddress'], zerotier_id)
+            )
```
Comment (ahussein) [inline JumpScale9Lib/clients/openvcloud/Space.py]: i think the way you are using the zerotier client is a bit strange, the client already offers an authorize api, you can check how its used here: https://github.com/Jumpscale/lib9/blob/development/JumpScale9Lib/clients/packetnet/PacketNetFactory.py#L194

File: JumpScale9Lib/clients/openvcloud/Space.py (hunk 2 above)
Comment (katia-e) [reply JumpScale9Lib/clients/openvcloud/Space.py]: @ahussein , Is it the same branch as in the end of url? Should they match?

Comment (ahussein) [reply JumpScale9Lib/clients/openvcloud/Space.py]: the one at the end of the url is to indicate that you want to run your node in development mode which means that you can access it without going through the node robot.

the one ipxe/<branch>/<zerotier_id>...is the branch of the zoero os that will be installed.

So i guess you need to different variables for that
- dev_mode = True
- branch = 'development'

File: JumpScale9Lib/clients/openvcloud/Space.py (hunk 3 above)
Comment (katia-e) [reply JumpScale9Lib/clients/openvcloud/Space.py]: yeah, good note, I'll redo it

File: JumpScale9Lib/clients/openvcloud/Space.py (hunk 2 above)
Comment (katia-e) [reply JumpScale9Lib/clients/openvcloud/Space.py]: @ahussein , if dev_mode = False, what do I insert instead of `development`?

Comment (katia-e) [reply JumpScale9Lib/clients/openvcloud/Space.py]: @ahussein also the goal of deploying this machine is talking to the node robot, so is that correct that I use development mode?

Comment (ahussein) [reply JumpScale9Lib/clients/openvcloud/Space.py]: If you run it in development mode, you can talk to the node directly or through the node robot. if it is not running in development mode, then we can only communicate with node through the robot node.

Yes, I think the default to dev_mode should be False, and if dev_mode is False then you just pass nothing, just omit the development at the end of the ipx command

**Pull request merged**